content/en/blog/_posts/2024-09-30-gateway-api-v1.2/index.md

```diff
@@ -0,0 +1,442 @@
+kind: Gateway
+metadata:
+  name: meshed-gateway
+  namespace: incoming
+spec:
+  gatewayClassName: meshed-gateway-class
+  listeners:
+  - name: http-listener
+    protocol: HTTP
+    port: 80
+  infrastructure:
+    labels:
+      istio-injection: enabled
+    annotations:
+      linkerd.io/inject: enabled
+```
+
+For more information, check out the [`infrastructure` API
+reference](https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.GatewayInfrastructure).
+
+### Backend protocol support
+
+Since Kubernetes v1.20, the Service and EndpointSlice resources have supported
+a stable `appProtocol`  field to allow users to specify the L7 protocol that
+Service supports. With the adoption of [KEP
+3726](https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/3726-standard-application-protocols),
+Kubernetes now supports three new `appProtocol` values:
+
+- `kubernetes.io/h2c` - HTTP/2 over cleartext as described in
+[RFC7540](https://www.rfc-editor.org/rfc/rfc7540)
+- `kubernetes.io/ws` - WebSocket over cleartext as described in
+[RFC6445](https://www.rfc-editor.org/rfc/rfc6445)
+- `kubernetes.io/wss` - WebSocket over TLS as described in
+[RFC6445](https://www.rfc-editor.org/rfc/rfc6445)
+
+With Gateway API 1.2.0, support for honoring `appProtocol` is now Standard.
+For example, given the following Service:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: websocket-service
+  namespace: my-namespace
+spec:
+  selector:
+    app.kubernetes.io/name: websocket-app
+  ports:
+    - name: http
+      port: 80
+      targetPort: 9376
+      protocol: TCP
+      appProtocol: kubernetes.io/ws
+```
+
+then an HTTPRoute that includes this Service as a `backendRef` will
+automatically upgrade the connection to use WebSockets rather than assuming
+that the connection is pure HTTP.
+
+For more information, check out
+[GEP-1911](https://gateway-api.sigs.k8s.io/geps/gep-1911/).
+
+## New additions to experimental channel
+
+### Named rules for *Route resources
+
+The `rules` field in HTTPRoute and GRPCRoute resources can now be named, in
+order to make it easier to reference the specific rule, for example:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: multi-color-route
+  namespace: faces
+spec:
+  parentRefs:
+    - name: my-gateway
+      kind: Gateway
+      port: 80
+  rules:
+  - name: center-rule
+    matches:
+    - path:
+        type: PathPrefix
+        value: /color/center
+    backendRefs:
+    - name: color-center
+      port: 80
+  - name: edge-rule
+    matches:
+    - path:
+        type: PathPrefix
+        value: /color/edge
+    backendRefs:
+    - name: color-edge
+      port: 80
+```
+
+Logging or status messages can now refer to these two rules as `center-rule`
+or `edge-rule` instead of being forced to refer to them by index. At present,
+rule names are only available for logging and status. For more information,
```

Comment:
> At present, rule names are only available for logging and status.

I would probably rephrase this part or remove it altogether.

GEP states that implementations MAY use the `name` field for other things as well.

If rephrasing it, maybe calling out that the field is optional to the implementations and therefore users are advised to check regarding support?

Reply:
Dropping, thanks for the callout!